### DISCUSSION TITLE
Why is VSCode intellisense/auto-import between packages using relative paths?

### DISCUSSION THREAD

**Discussion by colemahatmccreary** [vercel/turborepo#7629] [Help]
### Summary

I've set up a new monorepo following the quickstart guide, scaffolding from the 'basic' example.

After installing, I initialize a new package named @myrepo/types (```packages/types```), and export several TS types from it, including one named ```Transaction```. 

I then initialize a second package named @myrepo/lib (```packages/lib```), and add ```"@myrepo/types": "*"``` as a dependency.

In ```packages/lib/index.ts```, I then declare the following variable, without first importing ```Transaction``` manually:

```
let new_transaction: Transaction = {
     // relevant fields here
}
```

Following this pattern, I would expect intellisense to have displayed an option for auto-importing ```Transaction``` from ```@myrepo/types``` when annotating the ```new_transaction``` variable. Instead, it shows an option to auto-import from ```../types``` (using a relative path instead of "package syntax").

However if i manually write ```import {Transaction} from "@myrepo/types"```, things still work fine, so this seems to be an arbitrary choice made by VSCode.

I'd prefer the package import syntax, so how do I ensure intellisense/autoimport uses the local package syntax (```import {}  from "@myrepo/types"```) "by default rather than relative paths?

Am I missing something here?

**Comment by anthonyshew** [comment]
Do you have a reproduction we could look at?

**Reply by colemahatmccreary** [reply]
All I did was start from the template provided by create-turbo@latest, run npm init inside of two new directories (packages/types and packages/lib) and export a Transaction type from packages/types/index.ts. 

And the issue surfaced upon trying to import this type into packages/lib/index.ts (the package includes the other in dependencies as described above). 

I've done it multiple times from scratch to confirm I didn't add some other steps, but that's literally all there is to it.

**Reply by anthonyshew** [reply]
Can you provide the reproduction for that?

**Reply by skorphil** [reply]
@colemahatmccreary @anthonyshew 
Facing same problem. Auto-link suggests `../shared/jsfunction` instead of package import syntax `shared/jsfunction`

The simplest reproduction - pure js npm monorepo project https://github.dev/skorphil/npm-js-monorepo/blob/6cee5c1128bddec711a5c234edffffea243b91c7/packages/consumer/consumer.js#L2

![Screenshot 2024-05-11 at 14 10 28](https://github.com/vercel/turbo/assets/6762009/226cca16-b693-4d7f-8131-a1f8873f81ab)

**Reply by skorphil** [reply]
@colemahatmccreary @anthonyshew
When i created separate jsconfig.json for each of packages, autosuggestion stop working at all:
https://github.dev/skorphil/npm-js-monorepo/blob/f4e5ee026001283afcb3a3b178cd3cc191c36a32/packages/consumer/consumer.js#L1

**Comment by ScottMorse** [comment]
For others struggling with solutions to this, in [my SO answer here](https://stackoverflow.com/questions/69652809/how-can-i-get-vscode-auto-imports-working-in-a-typescript-monorepo/79301919#79301919) I detail how you can fix this  without declaration files or using the tsconfig `"paths"` by referencing workspaces by `"@example/workspace-name": "*"` in each dependent workspace's `package.json` `"dependencies"`. 

There are more details to the conditions for getting this working in the full answer in the link.

I also added `"typescript.preferences.includePackageJsonAutoImports": "on"` to my VSCode settings.json.
